docs/docs/icicle/rust-bindings/merkle.md

```diff
@@ -0,0 +1,310 @@
+
+# Merkle Tree API Documentation (Rust)
+
+This is the Rust version of the **Merkle Tree API Documentation** ([C++ documentation](../primitives/merkle.md)). It mirrors the structure and functionality of the C++ version, providing equivalent APIs in Rust.
+
+---
+
+## What is a Merkle Tree?
+
+A **Merkle tree** is a cryptographic data structure that allows for **efficient verification of data integrity**. It consists of:
```

Comment:
do you realy need to copy paste this section per API?

Reply:
I will link the rust to general part for the intro